scrapy/extensions/feedexport.py

```diff
@@ -328,8 +363,40 @@ def _get_instance(self, objcls, *args, **kwargs):
     def _get_exporter(self, file, format, *args, **kwargs):
         return self._get_instance(self.exporters[format], file, *args, **kwargs)
 
-    def _get_storage(self, uri):
-        return self._get_instance(self.storages[urlparse(uri).scheme], uri)
+    def _get_storage(self, uri, feed_options):
+        """Fork of create_instance specific to feed storage classes
+
+        It supports not passing the *feed_options* parameters to classes that
+        do not support it, and issuing a deprecation warning instead.
+        """
+        feedcls = self.storages[urlparse(uri).scheme]
+        crawler = getattr(self, 'crawler', None)
+
+        def build_instance(builder, *args):
+            argument_names = get_func_args(builder)
+            if 'feed_options' in argument_names:
+                _kwargs = {'feed_options': feed_options}
+            else:
+                warnings.warn(
+                    "{} does not support the 'feed_options' keyword argument"
+                    .format(builder.__qualname__),
+                    category=ScrapyDeprecationWarning
+                )
```

Comment:
This deprecation logic is repeated a few times, would it be possible to extract a helper?